client/java/build.gradle

```diff
@@ -45,12 +46,22 @@ configurations {
 }
 
 dependencies {
-    implementation "com.fasterxml.jackson.core:jackson-core:$jackson_version"
-    implementation "com.fasterxml.jackson.core:jackson-annotations:$jackson_version"
-    implementation "com.fasterxml.jackson.core:jackson-databind:$jackson_version"
-    implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:$jackson_version"
+    implementation "com.fasterxml.jackson.core:jackson-core:${jacksonVersion}"
+    implementation "com.fasterxml.jackson.core:jackson-annotations:${jacksonVersion}"
+    implementation "com.fasterxml.jackson.core:jackson-databind:${jacksonVersion}"
+    implementation "com.fasterxml.jackson.datatype:jackson-datatype-jdk8:${jacksonVersion}"
+    implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:${jacksonVersion}"
+    implementation 'com.google.code.findbugs:jsr305:3.0.2'
```

Comment:
Do we need this in runtime?

Reply:
I think we want to use `implementation`, instead of `api`? (see [API and implementation separation](https://docs.gradle.org/current/userguide/java_library_plugin.html))

Reply:
Yep, you're right.

Reply:
That lib `com.google.code.findbugs` introduces the package `javax.annotation.*`. Meaning, `com.google.code.findbugs` is the reference implementation of the `JSR-305`.

Reply:
Update: we actually want to use `compileOnly` and shade the libraries we're using.

`implementation` prevents compile-time dependency on the exposed library, but it still is required at runtime, and can conflict with other versions.